src/Pos/Wallet/Web/Tracking.hs

```diff
@@ -177,27 +178,28 @@ type WalletTrackingEnv ext ctx m =
      , WS.MonadWalletWebDB ctx m
      , MonadSlotsData m
      , WithLogger m
-     , HasLens GenesisUtxo ctx GenesisUtxo)
+     )
 
 syncWalletOnImportWebWallet :: WalletTrackingEnv ext ctx m => [EncryptedSecretKey] -> m ()
 syncWalletOnImportWebWallet = syncWalletsWithGState @WalletSscType
 
 txMempoolToModifierWebWallet :: WalletTrackingEnv ext ctx m => EncryptedSecretKey -> m CAccModifier
 txMempoolToModifierWebWallet encSK = do
-    let wHash (i, TxAux {..}) = WithHash taTx i
+    let wHash (i, TxAux {..}, _) = WithHash taTx i
         wId = encToCId encSK
         getDiff = const Nothing  -- no difficulty (mempool txs)
         getTs = const Nothing  -- don't give any timestamp
-    txs <- getLocalTxs
+    (txs, undoMap) <- getLocalTxsNUndo
+    let msgErr id = error $ sformat ("There is no undo corresponding to TxId #"%build%" from txp mempool") id
```

Comment:
We can perfectly do `throwM` with `logError` here instead of `error`, which seems much better.